components/gitpod-db/go/oidc_client_config.go

```diff
@@ -30,6 +31,27 @@ func (c *OIDCClientConfig) TableName() string {
 }
 
 type OIDCSpec struct {
+	OAuth2   OAuth2Config
+	Verifier VerifierConfig
+}
+
+type OAuth2Config struct {
+	// ClientID is the application's ID.
+	ClientID string `json:"clientId"`
+
+	// ClientSecret is the application's secret.
+	ClientSecret string `json:"clientSecret"`
+
+	// RedirectURL is the URL to redirect users going through
+	// the OAuth flow, after the resource owner's URLs.
+	RedirectURL string `json:"redirectUrl"`
+
+	// Scope specifies optional requested permissions.
+	Scopes []string `json:"scopes"`
+}
+
+type VerifierConfig struct {
+	ClientID string `json:"clientId"`
 }
```

Comment:
Feels wrong to have to define re-define all of these fields. However, I could not find a Go library which would include `json` annotations on the structs to guarantee the fields will remain consistent over time (and resilient to rename). If we find one, we can change this.

Reply:
Can you add this comment to the struct so we keep that in mind?

Reply:
done